### PR TITLE
Incompatible detector improvements

### PR DESCRIPTION
**Pull request by krzychu124**
- added CSM (Cities Skylines Multiplayer) to the list of incompatible mods
 - improved detection and removal of local mods, 
 - extended scanning with potentially game-breaking mods (CSM)

[Compiled build](https://ci.appveyor.com/api/projects/krzychu124/tmpe/artifacts/TMPE.zip?branch=incompatible-detector-improvements)

### CODE REVIEW

**Review by chameleon-tbn — APPROVED** [review]
lgtm - thanks for adding it... I have this question at least 2 Times a week in support.

File: TLM/TLM/Util/ModsCompatibilityChecker.cs [hunk 1]
```diff
@@ -23,9 +24,11 @@ public class ModsCompatibilityChecker {
 
         // parsed contents of incompatible_mods.txt
         private readonly Dictionary<ulong, string> knownIncompatibleMods;
+        private readonly HashSet<string> gameBreaking;
```
Comment (DaEgi01) [inline TLM/TLM/Util/ModsCompatibilityChecker.cs:27]: lol

File: TLM/TLM/UI/IncompatibleModsPanel.cs [hunk 2]
```diff
@@ -273,14 +291,25 @@ public class IncompatibleModsPanel : UIPanel {
         /// </summary>
         /// <param name="mod">The <see cref="PluginInfo"/> associated with the mod that needs deleting.</param>
         /// <returns>Returns <c>true</c> if successfully deleted, otherwise <c>false</c>.</returns>
-        private bool DeleteLocalTMPE(PluginInfo mod) {
+        private bool DeleteLocalMod(PluginInfo mod) {
```
Comment (DaEgi01) [inline TLM/TLM/UI/IncompatibleModsPanel.cs:294]: is it worth it to even have that code in TM:PE?
this is convenience stuff and imo it should be in a separate mod.

**Review by DaEgi01 — APPROVED** [review]

File: TLM/TLM/UI/IncompatibleModsPanel.cs (hunk 2 above)
Comment (krzychu124) [reply TLM/TLM/UI/IncompatibleModsPanel.cs:294]: You mean that particular method or Panel/Extension in general?

File: TLM/TLM/Util/ModsCompatibilityChecker.cs (hunk 1 above)
Comment (krzychu124) [reply TLM/TLM/Util/ModsCompatibilityChecker.cs:27]: If having a mod installed that breaks XML parsing in every other mod then what is it? 😄

File: TLM/TLM/UI/IncompatibleModsPanel.cs (hunk 2 above)
Comment (DaEgi01) [reply TLM/TLM/UI/IncompatibleModsPanel.cs:294]: I mean the whole associated functionality. I have not checked what code is calling it though.

File: TLM/TLM/Util/ModsCompatibilityChecker.cs (hunk 1 above)
Comment (DaEgi01) [reply TLM/TLM/Util/ModsCompatibilityChecker.cs:27]: pita xD
private readonly hashset<string> thisThingShouldDie =

File: TLM/TLM/UI/IncompatibleModsPanel.cs (hunk 2 above)
Comment (krzychu124) [reply TLM/TLM/UI/IncompatibleModsPanel.cs:294]: all is custom, reads incompatible mods from txt file (assembly resource) or that hashset above (if assembly name is known - required for local mods) 

I agree that might be useful for [Compatibility Report](https://steamcommunity.com/sharedfiles/filedetails/?id=2633433869)

Comment (DaEgi01) [reply TLM/TLM/UI/IncompatibleModsPanel.cs:294]: so .. yeah .. as long as we don't have a "compatibility report" mod that is the goto for the whole community it makes sense to have some functionality in TM:PE that checks for imcompatible mods and displays some popup.
What I mean is the code to immediately delete the incompatible mod. I would not put that into TM:PE. but i won't mind if it stays too :P

Comment (DaEgi01) [reply TLM/TLM/UI/IncompatibleModsPanel.cs:294]: anyway ... MERGE IT already!! :D

**Pull request merged**